packages/orama/src/methods/insert.ts

```diff
@@ -35,20 +35,35 @@ export async function insert(orama: Orama, doc: Document, language?: string, ski
     const actualType = typeof value
     const expectedType = indexablePropertiesWithTypes[key]
 
+    if (expectedType.endsWith('[]') && Array.isArray(value)) {
+      continue
+    }
+
     if (actualType !== expectedType) {
       throw createError('INVALID_DOCUMENT_PROPERTY', key, expectedType, actualType)
     }
   }
 
   for (const prop of indexableProperties) {
     const value = values[prop]
+    const expectedType = indexablePropertiesWithTypes[prop]
 
     if (typeof value === 'undefined') {
       continue
     }
 
     await orama.index.beforeInsert?.(orama.data.index, prop, id, value, language, orama.tokenizer, docsCount)
-    await orama.index.insert(orama.index, orama.data.index, prop, id, value, language, orama.tokenizer, docsCount)
+    await orama.index.insert(
+      orama.index,
+      orama.data.index,
+      prop,
+      id,
+      value,
+      expectedType,
```

Comment:
Note: remeber to update the `Components` page in the docs package to reflect this.

Reply:
Done